modules/redshift-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/redshift/Redshift.scala

```diff
@@ -58,26 +49,25 @@ object Redshift {
 
           override val requiresEventsColumns: Boolean = false
 
-          override def updateTable(migration: Migration): Block = {
-            val ddlFile = MigrationGenerator.generateMigration(migration, 4096, Some(schema))
-
-            val (preTransaction, inTransaction) = ddlFile.statements.foldLeft((NoPreStatements, NoStatements)) {
-              case ((preTransaction, inTransaction), statement) =>
-                statement match {
-                  case s @ AlterTable(_, _: AlterType) =>
-                    (Item.AlterColumn(Fragment.const0(s.toDdl)) :: preTransaction, inTransaction)
-                  case s @ AlterTable(_, _) =>
-                    (preTransaction, Item.AddColumn(Fragment.const0(s.toDdl), ddlFile.warnings) :: inTransaction)
-                  case _ => // We explicitly support only ALTER TABLE here; also drops BEGIN/END
-                    (preTransaction, inTransaction)
-                }
-            }
-
-            val target = SchemaKey(migration.vendor, migration.name, "jsonschema", migration.to)
-            Block(preTransaction.reverse, inTransaction.reverse, Entity.Table(schema, target))
+          override def updateTable(shredModel: ShredModel.GoodModel, currentSchemaKey: SchemaKey): Block = {
+            val preTransaction =
+              if (shredModel.migrationsOutTransaction(Some(currentSchemaKey)).nonEmpty)
+                Item.AlterColumn(
+                  Fragment.const0(shredModel.migrations.outTransactionToSql(shredModel.tableName, schema, Some(currentSchemaKey)))
```

Comment:
I think it would be safer to copy the SQL generating logic from schema-ddl to here. So that `const0` would correctly escape any issues in the variables for the query.

Also there is string interpolation verions of const0 which I find a bit neater `fr"<sql here>"`.